### PR TITLE
Change small caliber projectiles to instant hit

### PR DESCRIPTION
**Pull request by PunkPun**
Goal is to make damage more consistent, problems shown in #16038. Those bullet projectiles are already invisible to the human eye so the best solution is to make them instant hit. I didn't change the mobile flaks aa cannon as it has major balance implications. Currently it doesn't deal full damage to planes at longer range. This could be tackled in future balance changes

### CODE REVIEW

**Comment by Smittytron** [comment]
👎 
Makes perfect sense for bullets to have a speed. A ranger sitting still should naturally take more damage than a ranger on the move.

**Comment by pchote** [comment]
IIRC the original games used instant-hit projectiles for things like this, so this is potentially a case where we can both improve the gameplay behavior (if you believe the premise of the PR) *and* make things a bit more true to the original.

**Comment by matjaeck** [comment]
Ok. My first reaction was :-1: but when you say "think about it", let's think about it.

**Comment by Punsho** [comment]
Because of how pillbox bursts work ranger still takes much less damage while moving but thing is that damage is consistent. This change allows for more precise balancing

**Comment by matjaeck** [comment]
> Because of how pillbox bursts work ranger still takes much less damage while moving but thing is that damage is consistent.

This is true and I didn't notice the difference while doing some quick tests. My initial objection resulted from the assumption that moving targets would take the same damage as static targets, and I didn't see what we win by changing this. (I'll refrain from commenting on balance or game design topics from now on since I have absolutely no clue about this.)

**Comment by Punsho** [comment]
Bigger differences are seen on units with low health and as I said, this allows for more precise balancing. I'm currently on a path of improving core interactions by fixing small unnecessary inconsistencies. This is the first step in that march. The second step is to make all AttackFrontal units have some facing tolerance. Currently lack of it creates weird and illogical interactions (I'll make many examples when I'm done with this pr)

**Comment by abcdefg30** [comment]
So, what is the call here? Needs a rebase as well.

**Comment by Punsho** [comment]
Rebased. The problem I'm having with this pr is creating an inconsistency for mobile flak. Its projectiles  are invisible but not instant. We cannot make them instant as it would be a highly impactful balance change. If this is merged that inconsistency will have to be addressed in future balance changes. Other than that it fixes some debatable bugs I mentioned above and allows for more precise balancing

**Review by ghost — APPROVED** [review]
I didn't notice the changes and you have good arguments so this shouldn't block your fine-tuning.

**Comment by matjaeck** [comment]
@Orb370 Can you give this a review or note your thoughts on this?

**Comment by Punsho** [comment]
This doesn't have a big impact on lv's vs rifles as in most cases rifles already dealt full or almost full damage them. It only impacts rangers as rifles at max range hit ranger doing half damage. Since rifles only shoot max range if they're stationary and have vision support it isn't a significant change. Also most of the time #16486 lets ranger to get in close to infantry before a single bullet is fired

**Comment by Orb370** [comment]
Well, I asked Punsho to take a look at light vehicles vs rifles (see above). That was really my only concern.

Flak trucks is a balance issue but seems like we are sidestepping that for now.

Pillboxes are kind of weak these days already, with this change it might be worthwhile buffing them. But it shouldn't be game breaking, or at least prevent this from getting merged.

**Comment by matjaeck** [comment]
From Discord:

```
well, I spent 30 minutes running through things
It'd be very significant vs Yaks and Longbows (and obviously Migs)
It has a small impact vs Rangers
Rifles did less damage to rangers too
like as much as half as much
at max range
The pillbox will also be significant
Since it does basically no damage at max range
```

This seems to have more impact on balance than I thought, but I'm overall still confident that this is a step in the right direction to make things more responsive and predictable.

**Comment by Punsho** [comment]
Just as a note it isn't highly significant for pillbox since damage goes from very minor to minor against moving targets

**Comment by matjaeck** [comment]
Dismissing my review. I don't want that any of @Punsho's changes are merged without approval of trustworthy people with experience in balancing the game.

**Comment by Orb370** [comment]
After a lot of thought, I don't really like this change. I think removing this only simplifies game play with no real advantages.

There's no issues this really solves.

It removes some interesting game play such as light vehicles being able to avoid pillbox damage by continually moving.

I agree with Smitty "Makes perfect sense for bullets to have a speed. A ranger sitting still should naturally take more damage than a ranger on the move.".

**Comment by Punsho** [comment]
Welp, LV's will still take a lot less damage from pillboxes if moving. Bullets having speed is not mentioned anywhere in game and is only spread mouth to mouth. It is removing unnecessary complexity that doesn't bring engaging gameplay. Bullets have speed because of early limitations that we do not have now.

**Comment by Punsho** [comment]
Of course stats such as damage modifiers are not mentioned nor visible in game but people can easily test in debug how much does a thing do damage to another. Bullets having speed is such a subtle thing. If a target is moving you just do not know how much damage you're going to deal. I would agree to leave this in if it wasn't dependant on distance and just on if a target is moving or not

**Comment by GraionDilach** [comment]
@Punsho 

> I would agree to leave this in if it wasn't dependant on distance and just on if a target is moving or not

`GrantConditionOnMove` with a condition to trigger an additional armor which is listed upon the `Verses` value of bullets alone, say BulletsOnMove: 50. Since Verses defaults for unlisted armors are 100, this does not affect anything else. The layered armor system introduced years ago is exactly for usecases like this one.

**Comment by abcdefg30** [comment]
A lot of time has passed and a new release was published. Can we either revisit or close this?

**Comment by Punsho** [comment]
We may want to close this. It is likely going to come with the next balance update either way with mobile flak aa also receiving instant hit

**Comment by matjaeck** [comment]
Then just merge now?

**Comment by abcdefg30** [comment]
Then we can close this and will see what is up once the fighting over the next set of balance changes ended.

**Pull request closed without merging**
